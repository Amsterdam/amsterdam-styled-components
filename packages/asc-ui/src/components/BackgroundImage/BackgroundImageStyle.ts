import styled, { css } from 'styled-components'
import { clearFix } from 'polished'

export type Props = {
  source: string
  position?: string
  aspectRatio?: number
  repeat?: string
  size?: string
}

export default styled.div<Props>`
  width: 100%;
  background-image: url(${({ source }) => source});
  background-repeat: ${({ repeat }) => repeat};
  background-position: ${({ position }) => position};
  background-size: ${({ size }) => size};
  ${({ aspectRatio }) =>
    aspectRatio &&
    css`
      &::before {
        content: '';
        width: 1px;
        margin-left: -1px;
        float: left;
        height: 0;
        padding-top: ${aspectRatio}%;
      }
    `}

  ${clearFix()}
`
