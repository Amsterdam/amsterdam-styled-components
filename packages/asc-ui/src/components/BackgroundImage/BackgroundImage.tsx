import type { Props } from './BackgroundImageStyle'
import BackgroundImageStyle from './BackgroundImageStyle'

function BackgroundImage({ ...otherProps }: Props) {
  return <BackgroundImageStyle {...otherProps} />
}

BackgroundImage.defaultProps = {
  position: 'center',
  repeat: 'no-repeat',
  size: 'auto',
}

export default BackgroundImage
