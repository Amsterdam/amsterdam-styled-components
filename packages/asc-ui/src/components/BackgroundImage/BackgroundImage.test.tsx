import { render } from '@testing-library/react'
import BackgroundImage from './BackgroundImage'

describe('BackgroundImage', () => {
  it('should render', () => {
    const { container } = render(
      <BackgroundImage source="http://the-image-path" />,
    )
    expect(container.firstChild).toBeDefined()
  })
})
