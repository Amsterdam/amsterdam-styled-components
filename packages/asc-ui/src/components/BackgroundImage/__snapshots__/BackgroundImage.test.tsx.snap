// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BackgroundImage should render 1`] = `
.c0 {
  width: 100%;
  background-image: url(http://the-image-path);
  background-repeat: no-repeat;
  background-position: center;
  background-size: auto;
}

.c0::after {
  clear: both;
  content: "";
  display: table;
}

<div
  class="c0"
/>
`;
